src/sst/core/main.cc

```diff
@@ -1265,62 +1266,39 @@ main(int argc, char* argv[])
-        UnitAlgebra global_max_sync_data_size_ua(ua_buffer);
-
-        ua_buffer = format_string("%" PRIu64 "B", global_sync_data_size);
-        UnitAlgebra global_sync_data_size_ua(ua_buffer);
-
-        ua_buffer = format_string("%" PRIu64 "B", max_mempool_size);
-        UnitAlgebra max_mempool_size_ua(ua_buffer);
-
-        ua_buffer = format_string("%" PRIu64 "B", global_mempool_size);
-        UnitAlgebra global_mempool_size_ua(ua_buffer);
-
-        g_output.output("\n");
-        g_output.output("\n");
-        g_output.output("------------------------------------------------------------\n");
-        g_output.output("Simulation Timing Information (Wall Clock Times):\n");
-        g_output.output("  Build time:                      %f seconds\n", max_build_time);
-        g_output.output("  Run loop time:                   %f seconds\n", max_run_time);
-        g_output.output("  Total time:                      %f seconds\n", max_total_time);
-        g_output.output("\n");
-        g_output.output(
-            "Simulated time:                    %s\n", threadInfo[0].simulated_time.toStringBestSI().c_str());
-        g_output.output("\n");
-        g_output.output("Simulation Resource Information:\n");
-        g_output.output("  Max Resident Set Size:           %s\n", max_rss_ua.toStringBestSI().c_str());
-        g_output.output("  Approx. Global Max RSS Size:     %s\n", global_rss_ua.toStringBestSI().c_str());
-        g_output.output("  Max Local Page Faults:           %" PRIu64 " faults\n", local_max_pf);
-        g_output.output("  Global Page Faults:              %" PRIu64 " faults\n", global_pf);
-        g_output.output("  Max Output Blocks:               %" PRIu64 " blocks\n", global_max_io_out);
-        g_output.output("  Max Input Blocks:                %" PRIu64 " blocks\n", global_max_io_in);
-        g_output.output("  Max mempool usage:               %s\n", max_mempool_size_ua.toStringBestSI().c_str());
-        g_output.output("  Global mempool usage:            %s\n", global_mempool_size_ua.toStringBestSI().c_str());
-        g_output.output("  Global active activities:        %" PRIu64 " activities\n", global_active_activities);
-        g_output.output("  Current global TimeVortex depth: %" PRIu64 " entries\n", global_current_tv_depth);
-        g_output.output("  Max TimeVortex depth:            %" PRIu64 " entries\n", global_max_tv_depth);
-        g_output.output(
-            "  Max Sync data size:              %s\n", global_max_sync_data_size_ua.toStringBestSI().c_str());
-        g_output.output("  Global Sync data size:           %s\n", global_sync_data_size_ua.toStringBestSI().c_str());
-        g_output.output("------------------------------------------------------------\n");
-        g_output.output("\n");
-        g_output.output("\n");
+    if ( myRank.rank == 0 && (cfg.verbose() || cfg.print_timing() || cfg.timing_json() != "") ) {
+        std::unique_ptr<TimingOutput> timingOutput =
```

Comment:
Is there a reason to use std::unique_ptr here instead of just creating an instance of the object in this scope?  Are you just wanting to avoid allocating the TimingOutput object on the stack?